clients/src/main/java/org/apache/kafka/common/acl/AclOperation.java

```diff
@@ -22,6 +22,24 @@
 
 /**
  * Represents an operation which an ACL grants or denies permission to perform.
+ *
+ * Some operations imply other operations.
+ *
+ * ALLOW ALL implies ALLOW everything
+ * DENY ALL implies DENY everything
+ *
+ * ALLOW READ implies ALLOW DESCRIBE
+ * ALLOW WRITE implies ALLOW DESCRIBE
+ * ALLOW DELETE implies ALLOW DESCRIBE
+ *
+ * ALLOW ALTER implies ALLOW DESCRIBE
+ * DENY DESCRIBE implies DENY ALTER
```

Comment:
If DENY DESCRIBE implied DENY WRITE/READ/etc, it might break users' existing setups.  Since ALTER was previously unused, it is safe to do this here.